corehq/motech/repeaters/models.py

```diff
@@ -273,6 +273,28 @@ def update_failure_streak(self, attempt):
             self.failure_streak += 1
         self.save()
 
+    def is_connection_working(self):
+        """
+        Tries to determine whether the remote API is accepting payloads
+        based on past success.
+        """
+        three_months = timedelta(days=90)
+        if self.failure_streak > 10_000:
+            # 10,000 misses without a single hit.
+            return False
+        if datetime.utcnow() - self.started_at < three_months:
+            # Three months grace period to get it working
+            return True
+        if self.last_success_at is None:
+            if self.failure_streak == 0:
+                # Never succeeded, but never failed either: Nothing sent yet.
+                return True
+        else:
+            if datetime.utcnow() - self.last_success_at < three_months:
+                # Has succeeded at least once in the last 3 months
+                return True
+        return False
```

Comment:
nit: it's hard to reason about what would result in this final `False` being hit - what would you think of this?
```suggestion
        if self.last_success_at is None:
            # Never succeeded, but never failed either: Nothing sent yet.
            return self.failure_streak == 0:
        else:
            # Has succeeded at least once in the last 3 months
            return datetime.utcnow() - self.last_success_at < three_months:
```

Reply:
Oh that's much nicer!